pages/api/zapbye.ts

```diff
@@ -0,0 +1,100 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import jsforce from 'jsforce';
+
+const CONSTANTS = {
+      SALESFORCE_EMAIL: process.env.SALESFORCE_EMAIL,
+      SALESFORCE_PASSWORD: process.env.SALESFORCE_PASSWORD,
+      SALESFORCE_SECURITY_TOKEN: process.env.SALESFORCE_SECURITY_TOKEN,
+      LEMLIST_API_KEY: process.env.LEMLIST_API_KEY,
+      SALESFORCE_TASK_SUBJECT: 'Lemlist Email',
+      SALESFORCE_LEMLISTTASK_ID: '0123k0000014TCyAAM',
+      SALESFORCE_STATUS: 'Completed',
+      SALESFORCE_TASK_SUBTYPE: 'Task',
+      SALESFORCE_SOBJECT_TYPE: 'Task',
+      LEMLIST_CAMPAIGN_FETCH_COOLDOWN_TIME_IN_MS: 30000,
+};
+
+let salesforceConnection;
+let cachedCampaignsArray;
+let lastCampaignCheckedTimestamp;
+
+export default async (req: NextApiRequest, res: NextApiResponse): Promise<void> => {
+      const {
+            method,
+      } = req;
+
+      if (method != "POST") {
+            res.statusCode = 405;
+            res.json({ error: "Method Not Allowed" });
+            return;
+      }
+
+      if(!salesforceConnection || !salesforceConnection.accessToken) {
+            console.log("Attempting to log in!");
+            salesforceConnection = new jsforce.Connection();
+            await salesforceConnection.login(CONSTANTS.SALESFORCE_EMAIL, CONSTANTS.SALESFORCE_PASSWORD + CONSTANTS.SALESFORCE_SECURITY_TOKEN, function(err, res) {
+                  if (err) {
+                        salesforceConnection = null;
+                        res.statusCode = 500;
+                        res.body = err;
+                        return console.error(err);
+                  }
+            });
+      }
+      let lemlistPayload = req.body;
```

Comment:
Usually, we prevent using `let` unless necessary. Is a good practice to prevent mutations bugs. 
```suggestion
      const lemlistPayload = req.body;
```